files/image_config/interfaces/interfaces.j2

```diff
@@ -47,4 +47,6 @@ iface eth0 {{ 'inet' if prefix | ipv4 else 'inet6' }} static
 iface eth0 inet dhcp
 {% endif %}
 #
+source /etc/network/interfaces.d/*
```

Comment:
This command will fail if interfaces.d is empty

Reply:
command will fail in case interfaces.d itself doesn't exist, but I've created it explicitly.
```
dpkg -l | grep ifupdown ; ls -la /etc/network/interfaces.d/ ; cat /etc/network/interfaces | grep source ;  sudo systemctl restart interfaces-config
ii  ifupdown2                          1.0~git20170314-1              all          Network Interface Management tool similar to ifupdown
total 4
drwxr-xr-x 2 root root    3 Mar 14  2017 .
drwxr-xr-x 1 root root 4096 Feb 26 09:20 ..
source /etc/network/interfaces.d/*
```
and the status:
```
sudo journalctl -xe
Mar 04 09:07:59 sonic systemd[1]: Starting ifupdown2 networking initialization...
-- Subject: Unit networking.service has begun start-up
-- Defined-By: systemd
-- Support: https://www.debian.org/support
-- 
-- Unit networking.service has begun starting up.
Mar 04 09:07:59 sonic networking[20784]: networking: Configuring network interfaces
Mar 04 09:07:59 sonic kernel: IPv6: ADDRCONF(NETDEV_UP): eth0: link is not ready
Mar 04 09:07:59 sonic dhclient[20805]: DHCPDISCOVER on eth0 to 255.255.255.255 port 67 interval 8
Mar 04 09:07:59 sonic dhclient[20823]: DHCPDISCOVER on eth0 to 255.255.255.255 port 67 interval 7
Mar 04 09:08:00 sonic ntpd[4160]: Deleting interface #12 eth0, fe80::111:aaaa:aaaa:aaaa%2#123, interface stats: received=0, sent=0, dropped=0, active_time=82 sec
Mar 04 09:08:00 sonic ntpd[4160]: Deleting interface #13 eth0, 1.1.1.1#123, interface stats: received=0, sent=0, dropped=0, active_time=79 secs
Mar 04 09:08:01 sonic CRON[20841]: pam_unix(cron:session): session opened for user root by (uid=0)
Mar 04 09:08:01 sonic CRON[20842]: (root) CMD (/usr/sbin/logrotate /etc/logrotate.conf > /dev/null 2>&1)
Mar 04 09:08:01 sonic CRON[20841]: pam_unix(cron:session): session closed for user root
Mar 04 09:08:02 sonic kernel: igb 0000:02:00.0 eth0: igb: eth0 NIC Link is Up 1000 Mbps Full Duplex, Flow Control: RX/TX
Mar 04 09:08:02 sonic kernel: IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
Mar 04 09:08:03 sonic ntpd[4160]: Listen normally on 14 eth0 [fe80::111:aaaa:aaaa:aaaa%2]:123
Mar 04 09:08:06 sonic dhclient[20823]: DHCPDISCOVER on eth0 to 255.255.255.255 port 67 interval 13
Mar 04 09:08:06 sonic dhclient[20823]: DHCPREQUEST of 1.1.1.1 on eth0 to 255.255.255.255 port 67
Mar 04 09:08:06 sonic dhclient[20823]: DHCPOFFER of 1.1.1.1 from 1.1.1.2
Mar 04 09:08:06 sonic dhclient[20823]: DHCPACK of 1.1.1.1 from 1.1.1.2
Mar 04 09:08:06 sonic dhclient[20823]: bound to 1.1.1.1 -- renewal in 266 seconds.
Mar 04 09:08:06 sonic systemd[1]: Started ifupdown2 networking initialization.
-- Subject: Unit networking.service has finished start-up
-- Defined-By: systemd
-- Support: https://www.debian.org/support
-- 
-- Unit networking.service has finished starting up.
-- 
-- The start-up result is done.
Mar 04 09:08:06 sonic interfaces-config.sh[20741]: + ifdown lo
Mar 04 09:08:06 sonic interfaces-config.sh[20741]: + ifup lo
Mar 04 09:08:06 sonic systemd[1]: Started Update interfaces configuration.
-- Subject: Unit interfaces-config.service has finished start-up
-- Defined-By: systemd
-- Support: https://www.debian.org/support
-- 
-- Unit interfaces-config.service has finished starting up.
-- 
-- The start-up result is done.
Mar 04 09:08:06 sonic sudo[20738]: pam_unix(sudo:session): session closed for user root
Mar 04 09:08:07 sonic ntpd[4160]: Listen normally on 15 eth0 1.1.1.1:123
Mar 04 09:08:11 sonic sudo[20895]:    admin : TTY=pts/0 ; PWD=/home/admin ; USER=root ; COMMAND=/bin/journalctl -xe
Mar 04 09:08:11 sonic sudo[20895]: pam_unix(sudo:session): session opened for user root by admin(uid=0)
lines 1173-1217/1217 (END)
```